src/main/java/com/endava/internship/warmup/domain/service/ArrayProcessorWithForLoops.java

```diff
@@ -1,153 +1,327 @@
      * @throws IllegalArgumentException when parameters are outside of input index bounds
      */
     @Override
     public int[] copyValues(int[] input, int startInclusive, int endExclusive) throws IllegalArgumentException {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        if (startInclusive < 0 || startInclusive > endExclusive) {
+            throw new IllegalArgumentException("copy value out of bounds");
+        }
+        int arr[] = new int[endExclusive - startInclusive];
+        int j = 0;
+        for (int i = startInclusive; i < endExclusive; i++) {
+            arr[j] = input[i];
+            j++;
+        }
+        return arr;
     }
 
     /**
      * Replace even index values with their doubles and odd indexed elements with their negative
+     *
      * @param input non-null immutable array of ints
      * @return new array with changed elements
      */
     @Override
     public int[] replace(final int[] input) {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        int[] arr = new int[input.length];
+        for (int i = 0; i < input.length; i++) {
+            if (input[i] % 2 == 0) {
+                arr[i] = input[i] * 2;
+            } else if (input[i] % 2 != 0) {
+                arr[i] = -input[i];
+            }
+        }
+
+        return arr;
     }
 
     /**
      * Find the second max value in the array
+     *
      * @param input non-null immutable array of ints
      */
     @Override
     public int findSecondMax(final int[] input) {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        int[] fsmArr = new int[input.length];
+        for (int i = 0; i < fsmArr.length; i++) {
+            fsmArr[i] = input[i];
+        }
+        int[] fsmArr_final = Arrays.stream(fsmArr).distinct().toArray();
+        /*Arrays.sort(fsmArr);*/
+        /*int tv;
+        for(int i = 0;i< fsmArr.length;i++){
+            for(int j= i+1; j<fsmArr.length;i++){
+                if(fsmArr[i] > fsmArr[j]){
+                    tv = fsmArr[i];
+                    fsmArr[i] = fsmArr[j];
+                    fsmArr[j] = tv;
+                }
+            }
+        }*/
+        return fsmArr_final[fsmArr.length - 2];
     }
 
     /**
      * Return in reverse first negative numbers, then positive numbers from array
+     *
      * @param input non-null immutable array of ints.
      * @return example: input {3, -5, 4, -7, 2 , 9}
-     *                  result: {-7, -5, 9, 2, 4, 3}
+     * result: {-7, -5, 9, 2, 4, 3}
      */
     @Override
     public int[] rearrange(final int[] input) {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        int[] arrRearranged = new int[6];
+        int minus = 0;
+        int minus_array[] = new int[2];
+        int plus_array[] = new int[4];
+        int plus = 0;
+        int j = 0;
```

Comment:
please consider to give meaningful variable names

Reply:
Done